rebasehelper/tests/conftest.py

```diff
@@ -70,21 +71,26 @@ def mocked_spec_object(spec_attributes):
     spec = SpecFile.__new__(SpecFile)
     spec.save = lambda: None
     for attribute, value in spec_attributes.items():
+        if attribute == 'macros':
+            for macro, properties in value.items():
+                rpm.addMacro(macro, properties.get('value', ''))
+            macros = MacroHelper.dump()
+            for macro, properties in value.items():
+                for m in macros:
+                    if m['name'] == macro:
+                        for prop, v in properties.items():
+                            if prop != 'value':
+                                m[prop] = v
+            value = macros
         if attribute == 'spec_content' and isinstance(value, str):
             value = SpecContent(value)
         setattr(spec, attribute, value)
 
-    if 'macros' in spec_attributes:
-        for macro, properties in spec_attributes['macros'].items():
-            rpm.addMacro(macro, properties.get('value', ''))
-
-        spec.macros = MacroHelper.dump()
-        for macro, properties in spec_attributes['macros'].items():
-            for m in spec.macros:
-                if m['name'] == macro:
-                    for prop, value in properties.items():
-                        if prop != 'value':
-                            m[prop] = value
     if hasattr(spec, 'spec_content') and not hasattr(spec, 'tags'):
         spec.tags = Tags(spec.spec_content, spec.spec_content)
     return spec
+
+
+@pytest.fixture
+def mocked_spec_object_copy(mocked_spec_object):  # pylint: disable=redefined-outer-name
+    return copy.deepcopy(mocked_spec_object)
```

Comment:
This is a nitpick, but still. Why are these new fixtures below `pytest_collection_modifyitems`  hook and not below `spec_object` fixture?